lib/sisimai/bite/email/exim.rb

```diff
@@ -0,0 +1,522 @@
+        }x,
+        # deliver.c:5425|  new->message = US"Too many \"Received\" headers - suspected mail loop";
+        contenterror: %r/Too[ ]many[ ]["]Received["][ ]headers/x,
+      }
+
+      # retry.c:902|  addr->message = (addr->message == NULL)? US"retry timeout exceeded" :
+      # deliver.c:7475|  "No action is required on your part. Delivery attempts will continue for\n"
+      # smtp.c:3508|  US"retry time not reached for any host after a long failure period" :
+      # smtp.c:3508|  US"all hosts have been failing for a long time and were last tried "
+      #                 "after this message arrived";
+      # deliver.c:7459|  print_address_error(addr, f, US"Delay reason: ");
+      # deliver.c:7586|  "Message %s has been frozen%s.\nThe sender is <%s>.\n", message_id,
+      # receive.c:4021|  moan_tell_someone(freeze_tell, NULL, US"Message frozen on arrival",
+      # receive.c:4022|  "Message %s was frozen on arrival by %s.\nThe sender is <%s>.\n",
+      ReDelayed = %r{(?:
+         retry[ ]timeout[ ]exceeded
+        |No[ ]action[ ]is[ ]required[ ]on[ ]your[ ]part
+        |retry[ ]time[ ]not[ ]reached[ ]for[ ]any[ ]host[ ]after[ ]a[ ]long[ ]failure[ ]period
+        |all[ ]hosts[ ]have[ ]been[ ]failing[ ]for[ ]a[ ]long[ ]time[ ]and[ ]were[ ]last[ ]tried
+        |Delay[ ]reason:[ ]
+        |Message[ ].+[ ](?:has[ ]been[ ]frozen|was[ ]frozen[ ]on[ ]arrival[ ]by[ ])
+        )
+      }x
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Exim'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-Failed-Recipients']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Exim
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil if     mhead['from']    =~ ReE[:from]
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [272.9/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->